packages/next/src/server/app-render/use-flight-response.tsx

```diff
@@ -78,6 +79,25 @@ export async function flightRenderComplete(
   }
 }
 
+const decoder = new TextDecoder('utf-8', { fatal: true })
+
+/**
+ * This function will attempt to decode a Uint8Array as a UTF-8 string. If the
+ * data is not valid UTF-8 it will return null.
+ *
+ * @param value A Uint8Array that can contain arbitrary data.
+ */
+function tryDecodeAsUtf8String(
+  value: Uint8Array,
+  stream: boolean
+): string | null {
```

Comment:
You're really better off inlining this manually and do the different code paths based on code flow instead of the temporary value.

The compiler can't optimize this without knowing whether whatever `decoder.decode` is might return null.